File: solutions/php/difference-of-squares/1/DifferenceOfSquares.php
<?php

declare(strict_types=1);

function squareOfSum(int $max): int
{
    $sum = array_sum(range(1, $max));
    return $sum ** 2;
}

function sumOfSquares(int $max): int
{
    return array_sum(array_map(function($a){return $a ** 2;}, range(1, $max)));
}

function difference(int $max): int
{
    return squareOfSum($max) - sumOfSquares($max);
}

File: solutions/python/difference-of-squares/1/difference_of_squares.py
def square_of_sum(number: int) -> int:
    return sum(range(number+1)) ** 2


def sum_of_squares(number: int) -> int:
    rng = range(number+1)
    return sum(el ** 2 for el in rng)


def difference_of_squares(number: int) -> int:
    return square_of_sum(number) - sum_of_squares(number)
